### PR TITLE
Stripe UI

### PR DESCRIPTION
**Pull request by varuniyer**
Members Only UI now sends Stripe card token to backend and prevents users from registering without entering in credit card info

### CODE REVIEW

File: members_only/js/components/pages/Setup.js [hunk 1]
```diff
@@ -49,13 +50,24 @@ class Setup extends Component {
...
+        if (this.token == null) {
+            this.setState({
+                notificationText: "Payment Incomplete.",
+                notificationType: "danger"
+            });
+
+            event.preventDefault();
+            return;
+        }
+        console.log(this.token)
```
Comment (JacksonWeber) [inline members_only/js/components/pages/Setup.js]: Please remove Debug code before committing.
```suggestion
        
```

**Review by JacksonWeber — CHANGES_REQUESTED** [review]
Please compile using `npm run dev` and commit the statics.

File: members_only/js/components/pages/Setup.js (hunk 1 above)
Comment (varuniyer) [reply members_only/js/components/pages/Setup.js]: Done

File: members_only/js/components/pages/Setup.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import React, {Component} from 'react';
+import StripeCheckout from 'react-stripe-checkout';
```
Comment (JacksonWeber) [inline members_only/js/components/pages/Setup.js:2]: This dependency is not in package.json. Please commit it.

**Review by JacksonWeber — CHANGES_REQUESTED** [review]

File: members_only/js/components/pages/Setup.js [hunk 3]
```diff
@@ -147,7 +159,12 @@ class Setup extends Component {
                             </p>
                         </div>
 
-                        <div className="field">
+                        <StripeCheckout
+                            stripeKey="pk_test_OzmEbenGWueVpVlAQ54kHyGj00CekUS2fy"
+                            token={this.onToken}
+                        />
+            
+                <div className="field" style={{'margin-top': '20px'}}>
```
Comment (JacksonWeber) [inline members_only/js/components/pages/Setup.js]: Please fix indentation on this line.

**Review by JacksonWeber — CHANGES_REQUESTED** [review]

File: members_only/js/components/pages/Setup.js (hunk 3 above)
Comment (varuniyer) [reply members_only/js/components/pages/Setup.js]: Done

File: members_only/js/components/pages/Setup.js (hunk 2 above)
Comment (varuniyer) [reply members_only/js/components/pages/Setup.js:2]: Done

**Comment by JacksonWeber** [comment]
@varuniyer Just in case you missed it (as it was only a comment), you still need to compile the project with `npm run dev` and commit the statics. 

Excellent job with the other changes though!

**Comment by varuniyer** [comment]
@JacksonWeber why would we want to commit the statics? The UI works completely fine after running `npm run dev` once. Realistically, we're going to be running `npm run dev` on demo day before presenting, so there's really no need to keep statics up to date.

**Comment by JacksonWeber** [comment]
> @JacksonWeber why would we want to commit the statics? The UI works completely fine after running `npm run dev` once. Realistically, we're going to be running `npm run dev` on demo day before presenting, so there's really no need to keep statics up to date.

They need to be deployed, therefore they need to be in the repo. The reason it works on your machine is because you have the updated statics. Heroku will not compile these statics without them being in the repo.

**Comment by varuniyer** [comment]
I believe you do not need to manually upload the latest statics to Heroku, just updated application code.

**Comment by JacksonWeber** [comment]
> I believe you do not need to manually upload the latest statics to Heroku, just updated application code.

You are incorrect. https://github.com/david-fisher/320-S19-Track2/pull/199 As you can see, I've even encountered the same issue myself. I can confirm that it will not work if you do not compile and commit the statics.

**Comment by varuniyer** [comment]
Here's a relevant article: https://medium.com/quick-code/deploying-production-build-of-react-app-to-heroku-2548d8bf6936
> Note that if you run `npm run build` it will generate all production files in build directory. You can try that locally. No harm in that.

This implies that we don't need the statics. It wouldn't hurt to have them, but it isn't needed to deploy our React app to Heroku. So there is probably a way we can use Heroku and not need to constantly update statics.

**Comment by JacksonWeber** [comment]
> Here's a relevant article: https://medium.com/quick-code/deploying-production-build-of-react-app-to-heroku-2548d8bf6936

This is not how our repo is setup in Heroku. This is assuming that this is a nodeJS based server. This is a Python based project and the Procfile is not being modified, therefore we still need to commit the statics.


> Note that if you run `npm run build` it will generate all production files in build directory. You can try that locally. No harm in that.

We will not run anything on demo-day other than the server. This is not a valid suggestion, as we want to show off our development server.

> This implies that we don't need the statics. It wouldn't hurt to have them, but it isn't needed to deploy our React app to Heroku. So there is probably a way we can use Heroku and not need to constantly update statics.

Your entire argument is based on the assumption that our app is setup exactly the way that the article is written. Please commit the statics.

**Comment by varuniyer** [comment]
I disagree. Heroku knows our npm dependencies and has all the latest code needed to build the React app. It doesn't make sense that the server solution we use would affect whether or not our frontend can be built when deploying on Heroku. I believe there are configuration options in either our backend or in Heroku that we are not taking advantage of.

**Review by JacksonWeber — APPROVED** [review]

**Comment by JacksonWeber** [comment]
I compiled it myself and committed.Thank you for the work and the changes you made.

**Comment by alexhulbert** [comment]
@JacksonWeber 

I'd like to hop in here with a quick comment on version control best practices. In almost all circumstances, committing generated code to a source control repository is _very_ bad practice. The biggest risk you run when you put these files in a git repository is that they become out of sync with the source files they depend on. Additionally, any time a dependent file is modified, other files must be regenerated and attached to the same commit, which not only pollutes each commit's diff, but can cause confusion when reviewing code changes in teams with more than one or two contributors.

Hosting providers like Heroku are specifically designed with build scripts or "hooks," which remove the need to commit non-source files. These will automatically execute to generate any necessary static or generated files and can be configured to run at several points within a hosted instance's lifecycle. Any build system worth its weight, regardless of how it's designed, will have these features. For Heroku, you can read up on their build script system [here](https://devcenter.heroku.com/changelog-items/1557).

Maybe before you start lecturing your team members about what Heroku can and cannot do, you should read up on some basic DevOps principles.

**Comment by ForTheLoveOfDevOps** [comment]
I'm with @alexhulbert on this one

**Comment by JacksonWeber** [comment]
@alexhulbert I know that there is a better way to setup the repo. 

It's 5 days before this project is due, we are not going to properly implement new processes. This isn't a lecture on Heroku's capabilities, but specifically about the project needs.

I appreiciate the consideration for best practices, and normally I would wholeheartedly agree with you. However, seeing the time constraint, I'm most concerned with seeing each 
of our teams succeed on demo day.

**Comment by ForTheLoveOfDevOps** [comment]
@JacksonWeber 

I'd like to hop on this as well. I feel as though in a class that teaches about Software Engineering specifically, you should maybe do, you know, **Software Engineering** right? 

Even if we disregard that, adding to your teammates PR and merging, before he has a chance to respond is not how you work in teams.

**Pull request merged**